js-watcher/cache_collection.go

```diff
@@ -0,0 +1,103 @@
+package main
+
+import (
+	"context"
+	"log"
+
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+type CacheCollection map[string]*Cache 
+
+func (c *CacheCollection) synchronize(ctx context.Context) {
```

Comment:
I'm not sure why but a pointer receiver to a collection is not recommended - see https://github.com/golang/go/wiki/CodeReviewComments#receiver-type

Reply:
Np I will change it, I initially was scared of using a value receiver here because of potentially copying the caches. But the map has pointers to the caches as elements - so it is all good. The pointer addresses to the caches will be copied not the caches themselves.

Reply:
Whoops - I meant to a `map` obviously...